src/accelerate/accelerator.py

```diff
@@ -405,6 +448,21 @@ def prepare_model(self, model):
             model = torch.nn.parallel.DistributedDataParallel(
                 model, device_ids=[self.local_process_index], output_device=self.local_process_index, **kwargs
             )
+        elif self.distributed_type == DistributedType.FSDP:
+            from torch.distributed.fsdp.fully_sharded_data_parallel import CPUOffload
+            from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel as FSDP
+            from torch.distributed.fsdp.wrap import default_auto_wrap_policy
+
+            auto_wrap_policy = None
+            cpu_offload = None
+            if os.environ.get("FSDP_OFFLOAD_PARAMS", "false") == "true":
+                cpu_offload = CPUOffload(offload_params=True)
+
+            if int(os.environ.get("FSDP_MIN_NUM_PARAMS", "0")) > 0:
+                auto_wrap_policy = functools.partial(
+                    default_auto_wrap_policy, min_num_params=int(os.environ["FSDP_MIN_NUM_PARAMS"])
+                )
```

Comment:
Those two should also be added to the state, and we should get them from it here.

Reply:
Done.